doc/mgmt/SONiC_OC_MGMT_Interface_HLD.md

```diff
@@ -0,0 +1,259 @@
+
+## 1.1 Requirements
+
+![](http://10.59.132.240:9009/projects/csg_sonic/documentation/graphics/templates/test1.png)
+
+### 1.1.1 Functional Requirements
+
+1. Provide CLI, REST and GNMI support for configuring and displaying management interface attributes.
+2. Enhance existing implementation of interfaces yang to include management interface handling.
+3. Enhance existing top level show commands for interfaces to include management interface details too.
+
+
+### 1.1.2 Configuration and Management Requirements
+1. Provide CLI/GNMI/REST support for configuring Management interface attributes.
+2. Provide CLI/GNMI/REST support for show Management  interface attributes/parameters.
+
+
+### 1.1.3 Scalability Requirements
+N/A
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+Will be enhancing the management framework management backed code and transformer methods to add support for Management interface Handling.
+
+
+### 1.2.2 Container
+All code changes will be done in management-framework container.
+
+### 1.2.3 SAI Overview
+ N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+Manage/configure management interface via GNMI, REST and CLI interfaces
+
+## 2.2 Functional Description
+Provide GNMI and REST support for get/set of Management interface attributes and CLI config and show commands to manage Management interface.
+
+# 3 Design
+## 3.1 Overview
+1. Transformer common app owns the openconfig-interface.yang models (which  means no separate app module required for interfaces yang objects handling) Will be deleting the existing interface app module.
+2. Provide annotations for required objects in interfaces and respective augmented models so that transformer core and common app will take care of handling interfaces objects.
+3. Provide transformer methods as per the annotations defined for interfaces and respective augmented models to take care of model specific logics and validations.
+
+## 3.2 DB Changes
+N/A
+### 3.2.1 CONFIG DB
+No changes to database schema's just populate/read Config DB.
+### 3.2.2 APP DB
+No changes to database schema's just read APP DB for getting interface attributes.
+### 3.2.3 STATE DB
+No changes to database schema's just read state DB for getting interface state details.
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+No changes to database schema's just read COUNTER DB for getting interface counters information.
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+N/A.
+
+## 3.4 SyncD
+N/A.
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+Can be reference to YANG if applicable. Also cover GNMI here.
+List of yang models will be need to add support for management interface management.
+1. openconfig-if-ethernet.yang
+2. openconfig-if-ip.yang
+3. openconfig-interfaces.yang
+
+Supported yang objects and attributes:
+
+module: openconfig-interfaces
```

Comment:
This needs to be enclosed in a "diff" code block so that it's easier to read. Check out the raw markdown  text for the sonic-mgmt-framework wiki: 
https://github.com/project-arlo/sonic-mgmt-framework/wiki/YANG-Object-Reference/_edit

Reply:
Addressed